File: TicTacToe/TicTacToe_Node/components/player.ts
import { PlayingPiece } from "./playingPieces/playingPiece.ts";

export class Player {
  private _name: string;
  private _playingPiece: PlayingPiece;

  constructor(name: string, piece: PlayingPiece) {
    this._name = name;
    this._playingPiece = piece;
  }

  get name() {
    return this._name;
  }

  get piece() {
    return this._playingPiece;
  }
}

File: TicTacToe/TicTacToe_Go/components/player.go
package components

import "ticTacToe/components/playingPieces"

func GetNewPlayer(name string, piece playingPieces.IPlayingPiece) *Player {
	return &Player{
		name:         name,
		playingPiece: piece,
	}
}

type Player struct {
	name         string
	playingPiece playingPieces.IPlayingPiece
}

func (p *Player) GetName() string {
	return p.name
}

func (p *Player) GetPlayingPiece() playingPieces.IPlayingPiece {
	return p.playingPiece
}
